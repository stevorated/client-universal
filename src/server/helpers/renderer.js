import React from 'react'
import { renderToString } from 'react-dom/server'
import { Helmet } from 'react-helmet'
import { StaticRouter } from 'react-router'
import { Provider } from 'react-redux'
import { ApolloProvider } from 'react-apollo'
import serialize from 'serialize-javascript'
import { ServerStyleSheet } from 'styled-components'
import Layout from '../../shared/Routes/Layout'
import { config } from '@fortawesome/fontawesome-svg-core'
config.autoAddCss = false

const { API_BASE } = process.env
const linkToFavicon = `${API_BASE}/public_images/favicon.png`
export default function renderer(req, store, client, context) {

  const App = (
    <Provider store={store}>
      <ApolloProvider client={client}>
        <StaticRouter location={req.url} context={context}>
          <Layout />
        </StaticRouter>
      </ApolloProvider>
    </Provider>
  )

  const sheet = new ServerStyleSheet()
  const content = renderToString(sheet.collectStyles(App))
  const styles = sheet.getStyleTags()
  
  const helmet = Helmet.renderStatic()

  return `
  <!DOCTYPE html>
    <html>
      <head>
      <meta charset="UTF-8">
      <meta name="viewport" content="width=device-width, initial-scale=1.0">
      <meta name="840430c5e70010722d679edee63ea147c19430e1" content="e73a6ac6dc339154d6038998ef2574d65b12bf5e" />
      <script type="text/javascript">
        WebFontConfig = {
          google: { families: [ 'Sigmar One::latin' ] }
        };
        (function() {
          var wf = document.createElement('script');
          wf.src = ('https:' == document.location.protocol ? 'https' : 'http') +
            '://ajax.googleapis.com/ajax/libs/webfont/1/webfont.js';
          wf.type = 'text/javascript';
          wf.async = 'true';
          var s = document.getElementsByTagName('script')[0];
          s.parentNode.insertBefore(wf, s);
        })(); </script>
      <link href="https://fonts.googleapis.com/css?family=Sigmar+One&display=swap" rel="stylesheet" />
      <link href="https://use.fontawesome.com/releases/v5.9.0/css/svg-with-js.css" rel="stylesheet" />
      ${helmet.title.toString()}
      ${helmet.meta.toString()}
      <link rel="icon" href="${linkToFavicon}" type="image/png"/>
      <link rel="stylesheet" href="/styles/main.css"/>
      ${styles}
      </head>
      <body>
        <div id="root">${content}</div>
        <div id="portal"></div>
        <script>
          window.__INITIAL_STATE__ = ${serialize(store.getState())}
        </script>
        <script>
          window.__APOLLO_STATE__ = ${serialize(client.extract())}
        </script>
        <script src="/bundle.js"></script>
      </body>
    </html>
  
  `
}
