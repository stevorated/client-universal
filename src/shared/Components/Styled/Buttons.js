// import React from 'react'
// import { Button } from 'reactstrap'
// import styled from 'styled-components'

// export const MainColorBtn = styled(Button)`
//   background: #c9783e;
// `

// export const SmallMainColorBtn = () => <MainColorBtn size="sm" /> 