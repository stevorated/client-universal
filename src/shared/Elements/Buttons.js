import styled from 'styled-components'
import { Button } from 'reactstrap'
import { black, elevation, orange } from '../Utils'

export const RoundButton = styled(Button)`
/* position: absolute; */
background: ${black};
z-index: 100;
border-radius: 100%;
`

export const SquareButton = styled(Button)`
background: ${orange};
border: none;
transition: all .4s ease;
${elevation[2]};
&:hover {
  ${elevation[4]};
}
`