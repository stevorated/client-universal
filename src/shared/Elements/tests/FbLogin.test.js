import React from "react"
import { shallow } from "enzyme"
import renderer from "react-test-renderer"
import "jest-styled-components"

import { findByTestAttr } from "../../../tests/utils"
import { FbLogin } from "./../Buttons"

let component, store

const setupShallowRender = (props = {}) => {
  return shallow(<FbLogin {...props} />)
}

describe("<FbLogin />", () => {
  it("renders with login text correctly", () => {
    component = setupShallowRender()
    const button = findByTestAttr(component, "fb-login")
    expect(button.text()).toBe("<FontAwesomeIcon />login with Facebook")
  })
  it("renders with register text correctly", () => {
    component = setupShallowRender({ register: true })
    const button = findByTestAttr(component, "fb-login")
    expect(button.text()).toBe("<FontAwesomeIcon />register with facebook")
  })
})
