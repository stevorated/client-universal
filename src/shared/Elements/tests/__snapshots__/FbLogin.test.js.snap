// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FbLogin /> matches Snapshot 1`] = `
.c0 {
  box-shadow: 14px 14px 28px rgba(0,0,0,0.25),0 10px 10px rgba(0,0,0,0.22);
}

<a>
  <button
    className="big-btn c0"
    data-test="fb-login"
    style={
      Object {
        "background": "#3b5998",
        "color": "white",
      }
    }
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-facebook-square fa-w-14 fa-lg mr-3"
      data-icon="facebook-square"
      data-prefix="fab"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M400 32H48A48 48 0 0 0 0 80v352a48 48 0 0 0 48 48h137.25V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.27c-30.81 0-40.42 19.12-40.42 38.73V256h68.78l-11 71.69h-57.78V480H400a48 48 0 0 0 48-48V80a48 48 0 0 0-48-48z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    login with Facebook
  </button>
</a>
`;
